admin-dev/themes/new-theme/js/pages/product/components/combination-list/combinations-list-renderer.ts

```diff
@@ -0,0 +1,283 @@
+          this.$combinationsListContainer,
+        ).removeAttr('data-sort-is-current');
+        $(
+          CombinationsMap.sortableColumns,
+          this.$combinationsListContainer,
+        ).removeAttr('data-sort-direction');
+
+        // Set correct data in current column, we need to force the attributes for CSS matching
+        $sortableColumn.data('sortIsCurrent', 'true');
+        $sortableColumn.data('sortDirection', direction);
+        $sortableColumn.attr('data-sort-is-current', 'true');
+        $sortableColumn.attr('data-sort-direction', direction);
+
+        // Finally update list
+        this.sortListCallback(columnName, direction);
+      },
+    );
+  }
+
+  /**
+   * This container can be updated/replaced outside of this class so we can't keep a reference in the class
+   * instead we need to use this getter each time we need to get the container so that it always returns a valid
+   * just-in-time existing dom element.
+   */
+  private getCombinationsTable(): JQuery {
+    return $(CombinationsMap.combinationsTable);
+  }
+
+  /**
+   * @param {Array} combinations
+   *
+   * @private
+   */
+  private renderCombinations(combinations: Array<Record<string, any>>): void {
+    const $combinationsTable = this.getCombinationsTable();
+    const $combinationsTableBody = $(CombinationsMap.combinationsTableBody);
+
+    $combinationsTableBody.empty();
+    $combinationsTable.find(CombinationsMap.bulkSelectAll).prop('checked', false);
+
+    let rowIndex = 0;
+    combinations.forEach((combination: Record<string, any>) => {
+      const $row = $(this.getPrototypeRow(rowIndex, combination));
+
+      if (combination.is_default) {
+        // Init first default, and handle radio behaviour amongst lines
+        $(CombinationsMap.list.isDefault, $row).prop('checked', true);
+      }
+      this.updateByPriceImpactTaxExcluded($(CombinationsMap.list.priceImpactTaxExcluded, $row));
+
+      // JS event to allow external module to change the row, add listeners, ... before it is added
+      this.eventEmitter.emit(ProductEventMap.combinations.buildCombinationRow, {combination, $row});
+
+      $combinationsTableBody.append($row);
+      rowIndex += 1;
+    });
+
+    this.eventEmitter.emit(ProductEventMap.combinations.listRendered);
+  }
+
+  private getPrototypeRow(rowIndex: number, combination: Record<string, any>): string {
+    let rowTemplate: string = this.prototypeTemplate.replace(new RegExp(this.prototypeName, 'g'), rowIndex.toString());
+    Object.keys(combination).forEach((field: string) => {
+      if (typeof combination[field] === 'boolean') {
+        rowTemplate = rowTemplate.replace(new RegExp(`__${field}__`, 'g'), (combination[field] ? '1' : '0'));
+      } else {
+        rowTemplate = rowTemplate.replace(new RegExp(`__${field}__`, 'g'), combination[field]);
+      }
+    });
+
+    return rowTemplate;
+  }
+
+  private updateByPriceImpactTaxExcluded($priceImpactTaxExcluded: JQuery): void {
```

Comment:
These 2 methods are almost identical, looks like there is something to do. there